### ISSUE TITLE
the influence of batchsize with  Inplace-ABN

### ISSUE THREAD

**Issue by chaytonmin** [kwea123/CasMVSNet_pl#5]
Excelent work with  Inplace-ABN to reduce the memory cost. 

Now I want to train cascade with larger resolution of 1152*864 for dtu dataset to find out whether larger resolution will improve the 3D reconstruction performance. However, the batchsize can only be set as one for larger input. I want to ask you whether the batchsize of Inplace-ABN will influence the performance and have you ever trained cascade_pl with larger resolution of dtu dataset, will the lager resolution improve the 3D reconstruction performance of cascade_pl?

Another question is about the evaluations of dtu dataset with  accuracy and completeness. Why  for accuracy and completeness the lower is the better? In the paper of dtu dataset (http://roboimagedata2.compute.dtu.dk/data/text/multiViewCVPR2014.pdf), I find that ( there is a tradeoff between completeness and accuracy with [19] being the most accurate and [2] being the most complete.) in the paper. Therefore, I'm confused about the evaluations of dtu dataset with  accuracy and completeness whether the lower is the better in the most MVS papers.

Last question is about the process of images in testing phase. I find that you just resize the images to the size for testing while in RMVSNet they will first resize images to approximate size and then crop the images to needed size. What's the difference of the two processing ways and which one is better for random size for testing?

**Comment by kwea123** [comment]
1. No, I didn't train for larger resolution. I trained on 640x512 only. As for the batchsize, I have experimented 2 with one GPU (like in the readme and my pretrained model), also 4 with 8 GPUs (so 32 in total), but I didn't see significant performance gain/decrease; so I'd say that maybe the batch size is not that important w.r.t the performance

2. The metrics are lower the better: they are sort of point cloud distance between ground truth and prediction, so lower is better. And the tradeoff between precision and completeness comes basically from the point cloud size: if you have more points you are more complete but less precise (because of noisy points) and vice versa.

3. I do resizing because it's the simplest way; I think one advantage of cropping is based on the assumption that the foreground object is in the middle of the image, so cropping can exclude background regions where mvsnet performes badly, and result in a point cloud with less noise. However I don't think "cropping v.s. resizing" is carefully evaluated in any paper yet, we just all do what we think is the best.

**Comment by chaytonmin** [comment]
> 1. No, I didn't train for larger resolution. I trained on 640x512 only. As for the batchsize, I have experimented 2 with one GPU (like in the readme and my pretrained model), also 4 with 8 GPUs (so 32 in total), but I didn't see significant performance gain/decrease; so I'd say that maybe the batch size is not that important w.r.t the performance
> 2. The metrics are lower the better: they are sort of point cloud distance between ground truth and prediction, so lower is better. And the tradeoff between precision and completeness comes basically from the point cloud size: if you have more points you are more complete but less precise (because of noisy points) and vice versa.
> 3. I do resizing because it's the simplest way; I think one advantage of cropping is based on the assumption that the foreground object is in the middle of the image, so cropping can exclude background regions where mvsnet performes badly, and result in a point cloud with less noise. However I don't think "cropping v.s. resizing" is carefully evaluated in any paper yet, we just all do what we think is the best.

Thanks a lot!

**Comment by chaytonmin** [comment]
For training, all the papers resize and crop dtu images. Can I just resize  images to 640*512 for training (not first resize to 800*600 and then crop to 640*512)  as testing? Will the different ways for training influence the final performance?

**Comment by kwea123** [comment]
I think you can do just resizing. It should have only little to no effect.

**Comment by chaytonmin** [comment]
> I think you can do just resizing. It should have only little to no effect. Just be careful to change the intrinsics correctly.

Got it. Thanks!
